public/app/plugins/panel/geomap/GeomapPanel.tsx

```diff
@@ -354,28 +361,46 @@ export class GeomapPanel extends Component<Props, State> {
     if (!this.map) {
       return false;
     }
-    const selected = this.layers.findIndex((v) => v.UID === uid);
-    if (selected < 0) {
+    const current = this.byName.get(uid);
+    if (!current) {
       return false;
     }
-    const layers = this.layers.slice(0);
-    try {
-      let found = false;
-      const current = this.layers[selected];
-      const info = await this.initLayer(this.map, newOptions, current.isBasemap);
-      const group = this.map?.getLayers()!;
-      for (let i = 0; i < group?.getLength(); i++) {
-        if (group.item(i) === current.layer) {
-          found = true;
-          group.setAt(i, info.layer);
-          break;
-        }
+
+    let layerIndex = -1;
+    const group = this.map?.getLayers()!;
+    for (let i = 0; i < group?.getLength(); i++) {
+      if (group.item(i) === current.layer) {
+        layerIndex = i;
+        break;
       }
-      if (!found) {
-        console.warn('ERROR not found', uid);
+    }
+
+    // Special handling for rename
+    if (newOptions.name !== uid) {
+      if (!newOptions.name) {
+        newOptions.name = uid;
+      } else if (this.byName.has(newOptions.name)) {
         return false;
       }
-      layers[selected] = info;
+      console.log('Layer name changed', uid, '>>>', newOptions.name);
+      this.byName.delete(uid);
+
+      uid = newOptions.name;
+      this.byName.set(uid, current);
+    }
+
+    // Type changed -- requires full re-initalization
+    if (current.options.type !== newOptions.type) {
+      // full init
+    } else {
+      // just update options
```

Comment:
TODO -- (but not this PR!)

We currently do full initialization on any change.  We *should* only need to do that when the type changes